fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/expressions/Comparisons.java

```diff
@@ -693,10 +693,19 @@ public static String toPrintable(@Nullable Object value) {
         private final Type type;
         @Nonnull
         protected final Object comparand;
+        @Nullable
+        protected ListComparison listComparison;
```

Comment:
I suppose I probably should have mentioned this early, but something about this feels wrong. Like, instead of having a `SimpleComparison` have an optional `ListComparison`, it feels like the `IN` comparison should just use a `ListComparison` always (or as much as possible). I'm not actually sure what the right way to fix that is (if the QueryComponent isn't already using a `ListComparison` or something), but I figured I'd mention it.